realm/src/androidTest/java/io/realm/internal/JNIQueryTest.java

```diff
@@ -725,6 +729,12 @@ public void testMaximumDate() {
 
         assertEquals(new Date(10000), table.where().maximumDate(0));
 
+        // test for "#1510 Expand date range"
+
+        table.add(new Date(TimeUnit.SECONDS.toMillis(Integer.MAX_VALUE + 1L)));
```

Comment:
Any reason to add these? You already tested elsewhere that dates outside the current range could be added. I don't think we need to verify all methods working on date?